pkg/pillar/cmd/nodeagent/handletimers.go

```diff
@@ -101,6 +102,25 @@ func handleUpgradeTestValidation(ctxPtr *nodeagentContext) {
 	}
 }
 
+// when baseos upgrade is inprogress,
+// check if vault is accessible, and if not reset the node
+func handleRebootOnVaultLocked(ctxPtr *nodeagentContext) {
+	if ctxPtr.vaultOperational {
+		return
+	}
+	vaultCutOffTime := ctxPtr.globalConfig.GlobalValueInt(types.VaultReadyCutOffTime)
+	timePassed := ctxPtr.timeTickCount
+	if timePassed > vaultCutOffTime {
+		errStr := fmt.Sprintf("Exceeded time for vault to be ready %d by %d seconds, rebooting",
+			vaultCutOffTime, timePassed-vaultCutOffTime)
+		log.Errorf(errStr)
+		scheduleNodeReboot(ctxPtr, errStr, types.BootReasonFallback)
```

Comment:
I think you need to intreoduce a new BootReason here (including in the proto file) since it isn't just a fallback but a vault failure.

Reply:
Done